File: app/src/main/java/com/highcom/comicmemo/InputMemo.java
package com.highcom.comicmemo;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.CompoundButton;
import android.widget.EditText;
import android.widget.ToggleButton;

import androidx.core.content.ContextCompat;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.HashMap;
import java.util.Map;

/**
 * Created by koichi on 2016/08/07.
 */
public class InputMemo extends Activity implements CompoundButton.OnCheckedChangeListener {

    ListDataManager manager;
    private boolean isEdit;
    private long id;
    private long status;
    ToggleButton tbContinue;
    ToggleButton tbComplete;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.input_memo);

        // 渡されたデータを取得する
        Intent intent = getIntent();
        isEdit = intent.getBooleanExtra("EDIT", false);
        id = intent.getLongExtra("ID", -1);
        status = intent.getLongExtra("STATUS", 0);
        ((EditText)findViewById(R.id.editTitle)).setText(intent.getStringExtra("TITLE"));
        ((EditText)findViewById(R.id.editAuthor)).setText(intent.getStringExtra("AUTHOR"));
        ((EditText)findViewById(R.id.editNumber)).setText(intent.getStringExtra("NUMBER"));
        ((EditText)findViewById(R.id.editMemo)).setText(intent.getStringExtra("MEMO"));

        manager = ListDataManager.getInstance();

        // キャンセルボタン処理
        Button cancelBtn = (Button) findViewById(R.id.cancel);
        cancelBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View arg0) {
                finish();
            }
        });

        // 完了ボタン処理
        Button doneBtn = (Button) findViewById(R.id.done);
        doneBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View arg0) {
                // 入力データを登録する
                EditText editTitle = (EditText) findViewById(R.id.editTitle);
                EditText editAuthor = (EditText) findViewById(R.id.editAuthor);
                EditText editNumber = (EditText) findViewById(R.id.editNumber);
                Integer chgNumber = 0;
                if (!editNumber.getText().toString().equals("")) {
                    chgNumber = Integer.parseInt(editNumber.getText().toString());
                }
                EditText editMemo = (EditText) findViewById(R.id.editMemo);

                Map<String, String> data = new HashMap<String, String>();
                data.put("id", Long.valueOf(id).toString());
                data.put("title", editTitle.getText().toString());
                data.put("author", editAuthor.getText().toString());
                data.put("number", chgNumber.toString());
                data.put("memo", editMemo.getText().toString());
                data.put("inputdate", getNowDate());
                data.put("status", Long.valueOf(status).toString());
                // データベースに追加or編集する
                manager.setData(isEdit, data);
                // 詳細画面を終了
                finish();
            }
        });

        tbContinue = (ToggleButton) findViewById(R.id.toggleContinue);
        tbComplete = (ToggleButton) findViewById(R.id.toggleComplete);
        tbContinue.setOnCheckedChangeListener(this);
        tbComplete.setOnCheckedChangeListener(this);
        if (status == 0) {
            setEnableToggleContinue();
        } else {
            setEnableToggleComplete();
        }
    }

    @Override
    public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
        if (buttonView.getId() == R.id.toggleContinue) {
            setEnableToggleContinue();
        } else if (buttonView.getId() == R.id.toggleComplete) {
            setEnableToggleComplete();
        }
    }

    private void setEnableToggleContinue() {
        tbContinue.setTextColor(ContextCompat.getColor(getApplicationContext(), R.color.white));
        tbContinue.setBackgroundDrawable(ContextCompat.getDrawable(getApplicationContext(), R.drawable.toggle_select_button));
        tbComplete.setTextColor(ContextCompat.getColor(getApplicationContext(), R.color.blue));
        tbComplete.setBackgroundDrawable(ContextCompat.getDrawable(getApplicationContext(), R.drawable.toggle_unselect_button));
        status = 0;
    }

    private void setEnableToggleComplete() {
        tbContinue.setTextColor(ContextCompat.getColor(getApplicationContext(), R.color.blue));
        tbContinue.setBackgroundDrawable(ContextCompat.getDrawable(getApplicationContext(), R.drawable.toggle_unselect_button));
        tbComplete.setTextColor(ContextCompat.getColor(getApplicationContext(), R.color.white));
        tbComplete.setBackgroundDrawable(ContextCompat.getDrawable(getApplicationContext(), R.drawable.toggle_select_button));
        status = 1;
    }

    private String getNowDate(){
        Date date = new Date();
        final SimpleDateFormat sdf = new SimpleDateFormat("yyyy/MM/dd");
        return sdf.format(date);
    }
}

File: app/src/main/java/com/highcom/comicmemo/InputMemo.kt
package com.highcom.comicmemo

import android.app.Activity
import android.os.Bundle
import android.view.*
import android.widget.*
import androidx.core.content.ContextCompat
import java.text.SimpleDateFormat
import java.util.*

/**
 * Created by koichi on 2016/08/07.
 */
class InputMemo : Activity(), CompoundButton.OnCheckedChangeListener {
    var manager: ListDataManager? = null
    private var isEdit = false
    private var id: Long = 0
    private var status: Long = 0
    var tbContinue: ToggleButton? = null
    var tbComplete: ToggleButton? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.input_memo)

        // 渡されたデータを取得する
        val intent = intent
        isEdit = intent.getBooleanExtra("EDIT", false)
        id = intent.getLongExtra("ID", -1)
        status = intent.getLongExtra("STATUS", 0)
        (findViewById<View>(R.id.editTitle) as EditText).setText(intent.getStringExtra("TITLE"))
        (findViewById<View>(R.id.editAuthor) as EditText).setText(intent.getStringExtra("AUTHOR"))
        (findViewById<View>(R.id.editNumber) as EditText).setText(intent.getStringExtra("NUMBER"))
        (findViewById<View>(R.id.editMemo) as EditText).setText(intent.getStringExtra("MEMO"))
        manager = ListDataManager.instance

        // キャンセルボタン処理
        val cancelBtn = findViewById<View>(R.id.cancel) as Button
        cancelBtn.setOnClickListener { finish() }

        // 完了ボタン処理
        val doneBtn = findViewById<View>(R.id.done) as Button
        doneBtn.setOnClickListener { // 入力データを登録する
            val editTitle = findViewById<View>(R.id.editTitle) as EditText
            val editAuthor = findViewById<View>(R.id.editAuthor) as EditText
            val editNumber = findViewById<View>(R.id.editNumber) as EditText
            var chgNumber = 0
            if (editNumber.text.toString() != "") {
                chgNumber = editNumber.text.toString().toInt()
            }
            val editMemo = findViewById<View>(R.id.editMemo) as EditText
            val data: MutableMap<String, String> = HashMap()
            data["id"] = java.lang.Long.valueOf(id).toString()
            data["title"] = editTitle.text.toString()
            data["author"] = editAuthor.text.toString()
            data["number"] = chgNumber.toString()
            data["memo"] = editMemo.text.toString()
            data["inputdate"] = nowDate
            data["status"] = java.lang.Long.valueOf(status).toString()
            // データベースに追加or編集する
            manager!!.setData(isEdit, data)
            // 詳細画面を終了
            finish()
        }
        tbContinue = findViewById<View>(R.id.toggleContinue) as ToggleButton
        tbComplete = findViewById<View>(R.id.toggleComplete) as ToggleButton
        tbContinue!!.setOnCheckedChangeListener(this)
        tbComplete!!.setOnCheckedChangeListener(this)
        if (status == 0L) {
            setEnableToggleContinue()
        } else {
            setEnableToggleComplete()
        }
    }

    override fun onCheckedChanged(buttonView: CompoundButton, isChecked: Boolean) {
        if (buttonView.id == R.id.toggleContinue) {
            setEnableToggleContinue()
        } else if (buttonView.id == R.id.toggleComplete) {
            setEnableToggleComplete()
        }
    }

    private fun setEnableToggleContinue() {
        tbContinue!!.setTextColor(ContextCompat.getColor(applicationContext, R.color.white))
        tbContinue!!.setBackgroundDrawable(
            ContextCompat.getDrawable(
                applicationContext,
                R.drawable.toggle_select_button
            )
        )
        tbComplete!!.setTextColor(ContextCompat.getColor(applicationContext, R.color.blue))
        tbComplete!!.setBackgroundDrawable(
            ContextCompat.getDrawable(
                applicationContext,
                R.drawable.toggle_unselect_button
            )
        )
        status = 0
    }

    private fun setEnableToggleComplete() {
        tbContinue!!.setTextColor(ContextCompat.getColor(applicationContext, R.color.blue))
        tbContinue!!.setBackgroundDrawable(
            ContextCompat.getDrawable(
                applicationContext,
                R.drawable.toggle_unselect_button
            )
        )
        tbComplete!!.setTextColor(ContextCompat.getColor(applicationContext, R.color.white))
        tbComplete!!.setBackgroundDrawable(
            ContextCompat.getDrawable(
                applicationContext,
                R.drawable.toggle_select_button
            )
        )
        status = 1
    }

    private val nowDate: String
        private get() {
            val date = Date()
            val sdf = SimpleDateFormat("yyyy/MM/dd")
            return sdf.format(date)
        }
}